### PR TITLE
use eventtracking in place of track

### PR DESCRIPTION
**Pull request by danialmalik**
- Deprecate `track`'s `tracker.send` and use `eventtracking`'s `tracker.emit` calls.
- Update unit tests accordingly

### CODE REVIEW

File: common/djangoapps/track/views/__init__.py [hunk 1]
```diff
@@ -111,27 +111,14 @@ def server_track(request, event_type, event, page=None):
...
-    # Remove them from the event here since they are captured elsewhere.
-    shim.remove_shim_context(event)
-
-    log_event(event)
+    context_override = contexts.course_context_from_url(page)
+    context_override['username'] = username
+    context_override['event_source'] = 'server'
+    context_override['page'] = page
+
+    event_tracker = eventtracker.get_tracker()
```
Comment (ziafazal) [inline common/djangoapps/track/views/__init__.py:119]: @danialmalik we are neither using `server_track` nor `task_track` according to Fahad's PR. Then why need these changes?

File: lms/djangoapps/certificates/tests/test_webview_views.py [hunk 2]
```diff
@@ -1582,7 +1582,7 @@ def test_certificate_evidence_event_emitted(self):
         )
         response = self.client.get(test_url)
         self.assertEqual(response.status_code, 200)
-        actual_event = self.get_event()
+        actual_event = self.get_event(1)
```
Comment (ziafazal) [inline lms/djangoapps/certificates/tests/test_webview_views.py:1585]: In comments mention why we need event at index `1`

File: openedx/core/lib/tests/assertions/events.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import six
 
 
-def assert_event_matches(expected, actual, tolerate=None):
+def assert_event_matches(expected, actual, tolerate=None, string_payload_key=None):
```
Comment (ziafazal) [inline openedx/core/lib/tests/assertions/events.py:9]: not sure why we need these changes.

File: lms/djangoapps/courseware/tests/test_module_render.py [hunk 4]
```diff
@@ -2054,19 +2057,23 @@ def get_event_context(self, event_type, event):  # pylint: disable=unused-argume
             return {'content': 'test1', 'data_field': 'test2'}
 
         AsideTestType.get_event_context = get_event_context
-        context_info = self.handle_callback_and_get_context_info(mock_tracker, problem_display_name)
+        context_info = self.handle_callback_and_get_context_info(mock_tracker, problem_display_name, call_idx=4)
```
Comment (ziafazal) [inline lms/djangoapps/courseware/tests/test_module_render.py:2060]: we need to explain on code comments why we need to add `call_idx`

File: common/djangoapps/track/views/__init__.py (hunk 1 above)
Comment (danialmalik) [reply common/djangoapps/track/views/__init__.py:119]: Previously we just replaced the `server_track` and `task_track` calls with eventtracking's `tracker.emit` calls. But upon observing [this PR](https://github.com/edx/edx-platform/pull/7802/files#diff-e0483519f30846bafc1b92fb53f9aee5R325), we realized that we can follow the same steps and replace the `log_event` (which uses `tracker.send`) with eventtracking's call. This way we won't have to change EVERY server_track and task_track call.

File: openedx/core/lib/tests/assertions/events.py (hunk 3 above)
Comment (danialmalik) [reply openedx/core/lib/tests/assertions/events.py:9]: @ziafazal there were two tests that had ` "data": '"GET" :{}, "POST": {}`. While running those tests, those often failed due to the different positioning of `GET` and `POST` in the string. JSON comparison was needed. `assert_event_matches` provides the option to parse JSON data before comparison but it does the parsing only for the `event` field. 
when we changed the event, the `event` field got transformed into the `data` field and thus we need to parse `data` field now. 
This extra parameter is for this purpose.

File: openedx/core/lib/tests/assertions/events.py [hunk 5]
```diff
@@ -188,17 +188,19 @@ def block_indent(text, spaces=4):
...
+def parse_event_payload(event, string_payload_key=None):
     """
-    Given an event, parse the "event" field as a JSON string.
+    Given an event and a key, parse the field matching the `key` as a JSON string.
 
     Note that this may simply return the same event unchanged, or return a new copy of the event with the payload
     parsed. It will never modify the event in place.
     """
-    if 'event' in event and isinstance(event['event'], six.string_types):
+    string_payload_key = string_payload_key or 'event'
```
Comment (ziafazal) [inline openedx/core/lib/tests/assertions/events.py:198]: @danialmalik  can we simplify this 
```
payload_key = 'event' if 'data' in event else 'data'
if payload_key in event and isinstance(event[payload_key], six.string_types):
        event = event.copy()
        try:
            event[payload_key] = json.loads(event[payload_key])
        except ValueError:
            pass
```

File: common/djangoapps/track/views/__init__.py [hunk 6]
```diff
@@ -156,22 +143,14 @@ def task_track(request_info, task_info, event_type, event, page=None):
...
+
+    context_override = contexts.course_context_from_url(page)
+    context_override['username'] = request_info.get('username', 'unknown')
+    context_override['event_source'] = 'task'
+    context_override['page'] = page
+
+    with eventtracker.get_tracker().context('edx.course.task', context_override):
+        eventtracker.emit(name=event_type, data=data)
+
+    return HttpResponse('success')
```
Comment (ziafazal) [inline common/djangoapps/track/views/__init__.py:156]: @danialmalik 
why need to return HttpResponse?

Comment (danialmalik) [reply common/djangoapps/track/views/__init__.py]: My bad. Left it there by mistake.

**Pull request closed without merging**